### PR TITLE
Add ed25519 SSH Key support 

### PR DESCRIPTION
**Pull request by jkroepke**
RSA keys are problematic in newer linux distros and should be avoided if possible.

Azure supports the usage of ED25519 keys which should be always prefer over RSA keys.

Ref: https://azure.microsoft.com/en-us/updates/v2/ssh-key-support-for-linux-vms

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -55,11 +55,22 @@ variable "admin_password" {
...
 
+variable "admin_ssh_key_algorithm" {
+  description = "The algorithm to use for the admin SSH key pair. Valid values are `RSA` or `ED25519`."
+  default     = "ED25519"
+  type        = string
+
+  validation {
+    condition     = contains(["RSA", "ED25519"], var.admin_ssh_key_algorithm)
+    error_message = "Valid values for ssh_key_algorithm are `RSA` or `ED25519`."
+  }
```
Comment (rswrz) [inline variables.tf:71]: If you are doing variable validation, you should also do a test in `./tests/local/`. Or, in this case, you can skip the validation and let the Azure provider throw an error if a wrong `algorithm` is used.

File: variables.tf
Comment (rswrz) [inline variables.tf:1]: Please sort variables by name. You can use [tfsort](https://github.com/AlexNabokikh/tfsort) for this.

**Review by rswrz — CHANGES_REQUESTED** [review]
Thanks for this improvement. Please refer to our [Terraform Module Style Guide](https://github.com/cloudeteer/terraform-governance/blob/367-define-terraform-code-styleguide-and-develop-during-module-development-cloudeteerterraform-governance/docs/Development%20-%20Terraform%20Style%20Guide.md).

Also, please run `terraform-docs .` to update the `README.md` file.

File: variables.tf
Comment (jkroepke) [reply variables.tf:1]: nice! thanks for bring this to my attention.

**Review by rswrz — APPROVED** [review]
The _azurerm_ provider at v3 does not support `ssh-ed25519` SSH keys.

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 3.111"
    }
}

variables {
  authentication_type = "SSH"
}
```

```console
$ terraform init -test-directory=tests/remote
Initializing the backend...
Initializing modules...
- test.tests.remote.main.setup_tests in tests/remote
Initializing provider plugins...
- Finding hashicorp/random versions matching ">= 3.0.0, ~> 3.0"...
- Finding hashicorp/tls versions matching ">= 4.0.0, ~> 4.0"...
- Finding azure/azapi versions matching ">= 1.14.0, ~> 1.14"...
- Finding hashicorp/azurerm versions matching ">= 3.111.0, ~> 3.111"...
- Installing hashicorp/random v3.6.3...
- Installed hashicorp/random v3.6.3 (signed by HashiCorp)
- Installing hashicorp/tls v4.0.6...
- Installed hashicorp/tls v4.0.6 (signed by HashiCorp)
- Installing azure/azapi v1.15.0...
- Installed azure/azapi v1.15.0 (signed by a HashiCorp partner, key ID 6F0B91BDE98478CF)
- Installing hashicorp/azurerm v3.116.0...
- Installed hashicorp/azurerm v3.116.0 (signed by HashiCorp)

$ terraform test -test-directory=tests/remote
tests/remote/main.tftest.hcl... in progress
  run "setup_tests"... pass
  run "deploy_module_linux_ssh"... fail
╷
│ Error: - the provided ssh-ed25519 SSH key is not supported. Only RSA SSH keys are supported by Azure
│
│   with azurerm_linux_virtual_machine.this[0],
│   on r-vm.tf line 7, in resource "azurerm_linux_virtual_machine" "this":
│    7: resource "azurerm_linux_virtual_machine" "this" {
│
╵
tests/remote/main.tftest.hcl... tearing down
```

The time has come (pretty fast ;)) that we need to raise the minimum version to v4.1.

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 4.1"
    }
}

variables {
  authentication_type = "SSH"
}
```

```console
$ terraform init -test-directory=tests/remote
Initializing the backend...
Initializing modules...
- test.tests.remote.main.setup_tests in tests/remote
Initializing provider plugins...
- Finding hashicorp/random versions matching ">= 3.0.0, ~> 3.0"...
- Finding hashicorp/tls versions matching ">= 4.0.0, ~> 4.0"...
- Finding azure/azapi versions matching ">= 1.14.0, ~> 1.14"...
- Finding hashicorp/azurerm versions matching ">= 3.111.0, ~> 4.1"...
- Installing hashicorp/random v3.6.3...
- Installed hashicorp/random v3.6.3 (signed by HashiCorp)
- Installing hashicorp/tls v4.0.6...
- Installed hashicorp/tls v4.0.6 (signed by HashiCorp)
- Installing azure/azapi v1.15.0...
- Installed azure/azapi v1.15.0 (signed by a HashiCorp partner, key ID 6F0B91BDE98478CF)
- Installing hashicorp/azurerm v4.2.0...
- Installed hashicorp/azurerm v4.2.0 (signed by HashiCorp)

$ terraform test -test-directory=tests/remote
tests/remote/main.tftest.hcl... in progress
  run "setup_tests"... pass
  run "deploy_module_linux_ssh"... pass
tests/remote/main.tftest.hcl... tearing down
tests/remote/main.tftest.hcl... pass

Success! 2 passed, 0 failed.
```

We could keep the default key algorithm at RSA and stick with v3, but I think it is absolutely fine to go with v4 from here.

File: variables.tf [hunk 2]
```diff
@@ -54,8 +54,14 @@ variable "admin_password" {
   type        = string
 }
 
+variable "admin_ssh_key_algorithm" {
+  description = "The algorithm to use for the admin SSH key pair. Valid values are `RSA` or `ED25519`."
+  default     = "ED25519"
+  type        = string
+}
+
```
Comment (rswrz) [inline variables.tf:62]: Update name and description to indicate that this variable is only used when `admin_ssh_public_key` is not set, triggering SSH key generation when `authentication_type` is `SSH`.

**Review by rswrz — CHANGES_REQUESTED** [review]

**Review by rswrz — DISMISSED** [review]

**Comment by rswrz** [comment]
The remote test is slow and somehow broken. I improved it in #20 and merged this changed here. Hopefully, now the test runs successfully.

**Pull request merged**